pydm/widgets/byte.py

```diff
@@ -0,0 +1,325 @@
+from ..PyQt.QtGui import QWidget, QTabWidget, QColor, QPen, QGridLayout, QLabel, QPalette, QFontMetrics, QPainter, QBrush, QStyleOption, QStyle
+from ..PyQt.QtCore import pyqtSignal, pyqtSlot, pyqtProperty, Qt, QStringList, QSize, QPoint
+from .channel import PyDMChannel
+import numpy as np
+
+class PyDMBitIndicator(QWidget):
+  def __init__(self, parent=None):
+    super(PyDMBitIndicator, self).__init__(parent)
+    self.setAutoFillBackground(True)
+    self.circle = False
+    self._painter = QPainter()
+    self._brush = QBrush(Qt.SolidPattern)
+    self._pen = QPen(Qt.SolidLine)
+    
+  def paintEvent(self, event):
```

Comment:
I could go either way, but in this case, the indicator is so simple that keeping it separate isn't really duplicating too much code.  The drawing widgets do so much more stuff that I don't think the indicator needs, and I think overriding the drawing class behavior to do the simple color changes I need would be more complicated than just doing it from scratch.